### PR TITLE
added verification for openapi-spec

### PR DESCRIPTION
**Pull request by EraKin575**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: #5679 

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by EraKin575** [comment]
@Shelley-BaoYue please review this PR

**Comment by EraKin575** [comment]
@Shelley-BaoYue hi! Is this the expected behavior from the new verification?

**Review by fisherxu — COMMENTED** [review]
Plz squash the 4 commits to1 commit, thanks :)
And also plz submit the updated file swagger.json.

File: apidoc/generated/openapi/zz_generated.openapi.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (fisherxu) [inline apidoc/generated/openapi/zz_generated.openapi.go:1]:
```suggestion
```

File: hack/verify-apidoc.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ret=0
+diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
+
+if [[ $ret -eq 0 ]]; then
+  echo "${DIFFROOT} up to date."
+else
+  echo "${DIFFROOT} is out of date. Please re-generate swagger.json"
+  exit 1 
+
```
Comment (fisherxu) [inline hack/verify-apidoc.sh:58]:
```suggestion
```

File: hack/verify-apidoc.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+trap "cleanup" EXIT SIGINT
+
+cleanup
+
+DIFFROOT="${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/api"
+TMP_DIFFROOT="${KUBEEDGE_ROOT}/_tmp/api"
+mkdir -p "${TMP_DIFFROOT}"
+cp -a "${DIFFROOT}"/* "${TMP_DIFFROOT}"
+
+sudo bash "${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/api/apidoc/tools/update-swagger-docs.sh"
```
Comment (Shelley-BaoYue) [inline hack/verify-apidoc.sh:39]: Seems we need first run [generate-openapi.sh](https://github.com/kubeedge/kubeedge/blob/master/staging/src/github.com/kubeedge/api/apidoc/tools/generate-openapi.sh) and then run update-swagger-docs.sh

File: hack/verify-apidoc.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+echo "diffing ${DIFFROOT} against freshly generated codegen"
+ret=0
+diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
+if [[ $ret -eq 0 ]]
+then
+  echo "${DIFFROOT} up to date."
+else
+  echo "${DIFFROOT} is out of date. Please run hack/update-codegen.sh"
```
Comment (Shelley-BaoYue) [inline hack/verify-apidoc.sh:50]: hack/update-codegen.sh need to be replaced

Comment (EraKin575) [reply hack/verify-apidoc.sh:50]: thanks for the review

File: hack/OWNERS [hunk 5]
```diff
@@ -8,5 +8,6 @@ reviewers:
   - fisherxu
   - kadisi
   - kevin-wangzefeng
+  - luomengY
```
Comment (Shelley-BaoYue) [inline hack/OWNERS:11]: Seems some other codes are introduced in this PR. Please check it, and it's recommend to squash all commits to one, and name the commit like `add verification for openapi doc`. Now the commit name `parent` cannot let others to known what work has done in this commit.

**Pull request closed without merging**